drivers/watchdog/wdt_dw.c

```diff
@@ -0,0 +1,239 @@
+#endif
+
+	/* Enable watchdog */
+	dw_wdt_enable(dev_config->base);
+	dw_wdt_counter_restart(dev_config->base);
+
+	return 0;
+}
+
+static int dw_wdt_disable(const struct device *dev)
+{
+	return -ENOTSUP;
+}
+
+static int dw_wdt_install_timeout(const struct device *dev, const struct wdt_timeout_cfg *config)
+{
+	const struct dw_wdt_dev_cfg *const dev_config = dev->config;
+	struct dw_wdt_dev_data *const dev_data = dev->data;
+	uint64_t period64;
+	uint32_t period;
+
+#if WDT_DW_INTERRUPT_SUPPORT
+	if (config->callback && !dev_config->irq_config) {
+#else
+	if (config->callback) {
+#endif
+		LOG_ERR("Interrupt is not configured, can't set a callback.");
+		return -ENOTSUP;
+	}
+
+	/* Window timeout is not supported by this driver */
+	if (config->window.min) {
+		LOG_ERR("Window timeout is not supported.");
+		return -ENOTSUP;
+	}
+
+	if (config->flags) {
+		LOG_ERR("Watchdog behavior is not configurable.");
+		return -ENOTSUP;
+	}
+
+	period64 = (uint64_t)dev_config->clk_freq * config->window.max;
+	period64 /= 1000;
+	if (!period64 || (period64 >> 31)) {
+		LOG_ERR("Window max is out of range.");
+		return -EINVAL;
+	}
+
+	period = period64 - 1;
+	period = ilog2(period);
+
+	if (period >= dw_wdt_cnt_width_get(dev_config->base)) {
+		LOG_ERR("Watchdog timeout too long.");
+		return -EINVAL;
+	}
+
+	/* The minimum counter value is 64k, maximum 2G */
+	dev_data->config = WDT_DW_FLAG_CONFIGURED | (period >= 15 ? period - 15 : 0);
+	dev_data->callback = config->callback;
+	return 0;
+}
+
+static int dw_wdt_feed(const struct device *dev, int channel_id)
+{
+	const struct dw_wdt_dev_cfg *const dev_config = dev->config;
+
+	/* Only channel 0 is supported */
+	if (channel_id)
+		return -EINVAL;
+
+	dw_wdt_counter_restart(dev_config->base);
+
+	return 0;
+}
+
+static const struct wdt_driver_api dw_wdt_api = {
+	.setup = dw_wdt_setup,
+	.disable = dw_wdt_disable,
+	.install_timeout = dw_wdt_install_timeout,
+	.feed = dw_wdt_feed,
+};
+
+static int dw_wdt_init(const struct device *dev)
+{
+	const struct dw_wdt_dev_cfg *const dev_config = dev->config;
+
+	/* Check component type */
+	if (dw_wdt_comp_type_get(dev_config->base) != WDT_COMP_TYPE_VALUE) {
+		LOG_ERR("Invalid component type %x", dw_wdt_comp_type_get(dev_config->base));
+		return -ENODEV;
+	}
+
+	dw_wdt_reset_pulse_length_set(dev_config->base, dev_config->reset_pulse_length);
+
+#if WDT_DW_INTERRUPT_SUPPORT
+	if (dev_config->irq_config)
+		dev_config->irq_config();
+#endif
+
+	return 0;
+}
+
+#if WDT_DW_INTERRUPT_SUPPORT
+static void dw_wdt_isr(const struct device *dev)
+{
+	const struct dw_wdt_dev_cfg *const dev_config = dev->config;
+	struct dw_wdt_dev_data *const dev_data = dev->data;
+
+	if (dw_wdt_interrupt_status_register_get(dev_config->base)) {
+		dw_wdt_clear_interrupt(dev_config->base);
+
+		if (dev_data->callback)
+			dev_data->callback(dev, 0);
+	}
+}
+#endif
+
+#define CHECK_RESET_PULSE_LENGTH_VALUE(rpl)                                                        \
```

Comment:
This macro name is pretty confuse, it is just if the pulse length value is valid right ?

Reply:
I removed the value check because it is no longer needed since using enum in yaml.